x-pack/plugins/infra/public/components/logging/log_text_stream/log_entry_message_column.tsx

```diff
@@ -90,3 +100,16 @@ const formatMessageSegments = (
 
     return 'failed to format message';
   });
+
+const renderMessageSegments = (messageSegments: LogMessagePart[]): string => {
+  return messageSegments
+    .map((messageSegment) => {
+      if (isConstantSegment(messageSegment)) {
+        return messageSegment.constant;
+      }
+      if (isFieldSegment(messageSegment)) {
+        return messageSegment.value.toString();
```

Comment:
:thought_balloon: This might make messages with complex values impossible to render correctly. But that's hard to avoid short of exposing the segment abstraction to the `render()` method.

What do you think about passing the constants and unstringified values as an array instead? Would that complicate the API too much?

Reply:
Do you remember on the top of your head any cases where one of the parts is a complex object? (Otherwise I will check the message formation rules myself tomorrow).

An option can be to test for the type of the value, and if it's anything other than a string use `JSON.stringify()` on it.

Reply:
I scanned all the test cases for the custom message formatters but didn't find anything where a complex field value was embedded in the message. :shrug: So maybe my concern was not justified.